inc/Models/SystemModel.h

```diff
@@ -138,6 +151,30 @@ namespace PCOE {
         virtual output_type outputEqn(const double t,
                                       const state_type& x,
                                       const noise_type& n) const = 0;
+        
+        /**
+         * Calculate event state.
+         *
+         * @param x  The model state vector at the current time step.
+         * @return   The
+         */
+        virtual event_state_type eventStateEqn(const state_type& x) const {
+            return event_state_type();
+        }
+        
+        /** Calculate observables of the model. Observables are those
+         * that are not measured, but are interested in being predicted for
+         * prognostics.
+         *
+         * @param t  Time
+         * @param x  The model state vector at the current time step.
+         * @return   The model output vector at the next time step.
+         **/
+        virtual observables_type observablesEqn(double t,
```

Comment:
I'm almost positive this doesn't have a counterpart in any MATLAB code. Can we please call it `calculateObservables`?

Reply:
I understand your argument- but I think the inconsistency of having all the meat of the model called *Eqn except for one function would add more confusion than the more descriptive name calculateObservables would resolve.